### ISSUE TITLE
If not starting with a quiz ID, forgot-password process redirects to blank quiz

### ISSUE THREAD

**Issue by arthurattwell** [electricbookworks/betterquiz#9]
This will be a rare bug, but may occur when a user happens to arrive at a Betterquiz site without a quiz ID. i.e. landing at `http://quiz.bettercare.co.za`, not `http://quiz.bettercare.co.za/quiz.php?qz=1`

To reproduce:
1. Go to http://quiz.bettercare.co.za/
2. Click 'Forgot password'
3. Enter registered email address
4. Click 'Forgot password'
5. Site provides new password. Login on same screen with new password.
6. Blank screen at `http://quiz.bettercare.co.za/quiz.php?qz=0`
7. Also, no new-password email is received.

A problem here, I assume, is that we have no catch-all landing page. That is, something to use other than `/quiz.php?qz=0` when the user has not arrived from a particular quiz. Should we default to something like a 404 message, such as `Sorry, we don't know which quiz you're looking for.` with a link to 'Profile'?

**Comment by craigmj** [comment]
Yes, the issue here is that we don't really know what to do if we don't know the user's required quiz. I like the 'Sorry, we don't know' idea. I will implement that on the question page, in the circumstance where we can't find the quiz for the user.
